### PR TITLE
Testing/1.16.1

### PR DESCRIPTION
**Pull request by jan-grimo**
**NOTE**: This is a PR meant for a new version/branch of the mongo-c-driver, not the branch as submitted.

See bincrafters/community#1132

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 from bincrafters import build_template_default
 
 if __name__ == "__main__":
-
-    builder = build_template_default.get_builder()
-
+    builder = build_template_default.get_builder(
+        pure_c=True,
+        build_policy="missing"
```
Comment (Croydon) [inline build.py]: Please don't do this. If prebuilds are missing let us know so we can fix it

**Review by Croydon — CHANGES_REQUESTED** [review]

File: build.py (hunk 1 above)
Comment (jan-grimo) [reply build.py]: I've reverted it. There seems to be no ICU build [for linux gcc 9](https://travis-ci.com/bincrafters/conan-mongo-c-driver/jobs/288331658).

**Comment by jan-grimo** [comment]
There also seem to be issues consuming https://github.com/bincrafters/conan-icu with clang and on windows.

I've taken the lazy approach here and simply disabled the use of ICU by default.

File: conanfile.py [hunk 2]
```diff
@@ -62,25 +80,33 @@ def build(self):
...
+
         self.cpp_info.includedirs = [os.path.join("include", "libmongoc-1.0"),
                                      os.path.join("include", "libbson-1.0")]
 
+        if self.options.icu and not self.options.shared:
+            self.cpp_info.libs.extend(self.deps_cpp_info["icu"].libs)
+
         if tools.os_info.is_macos:
-            self.cpp_info.exelinkflags = ['-framework CoreFoundation', '-framework Security']
+            self.cpp_info.exelinkflags = [
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
            self.cpp_info.frameworks.extend(["CoreFoundation", "Security"])
```

File: conanfile.py [hunk 3]
```diff
@@ -62,25 +80,33 @@ def build(self):
...
+                '-framework CoreFoundation', '-framework Security']
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
```
https://docs.conan.io/en/latest/howtos/link_apple_framework.html

File: conanfile.py [hunk 4]
```diff
@@ -62,25 +80,33 @@ def build(self):
...
             self.cpp_info.sharedlinkflags = self.cpp_info.exelinkflags
 
         if tools.os_info.is_linux:
-            self.cpp_info.libs.extend(["rt", "pthread"])
+            self.cpp_info.libs.extend(["rt", "pthread", "dl"])
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
            self.cpp_info.sytem_libs.extend(["rt", "pthread", "dl"])
```
https://docs.conan.io/en/latest/reference/conanfile/attributes.html#cpp-info

File: conanfile.py [hunk 5]
```diff
@@ -89,4 +115,5 @@ def package_info(self):
                 self.cpp_info.libs.append('resolv')
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
                self.cpp_info.system_libs.append('resolv')
```

File: conanfile.py [hunk 6]
```diff
@@ -89,4 +115,5 @@ def package_info(self):
...
 
             if tools.os_info.is_windows:
-                self.cpp_info.libs.extend(['ws2_32.lib', 'secur32.lib', 'crypt32.lib', 'BCrypt.lib', 'Dnsapi.lib'])
+                self.cpp_info.libs.extend(
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
                self.cpp_info.system_libs.extend(
```

File: conanfile.py [hunk 7]
```diff
@@ -36,23 +37,40 @@ def requirements(self):
...
-        cmake.definitions["ENABLE_EXAMPLES"] = False
-        cmake.definitions["ENABLE_AUTOMATIC_INIT_AND_CLEANUP"] = False
-        cmake.definitions["ENABLE_BSON"] = "ON"
-        cmake.definitions["ENABLE_SASL"] = "OFF"
-        cmake.definitions["ENABLE_STATIC"] = "OFF" if self.options.shared else "ON"
-        cmake.definitions["ENABLE_SHM_COUNTERS"] = "OFF"
+        # There are several options that would autodetect optional dependencies
+        # and add them by default, need to disable all of those to ensure
+        # consistent abi and dependencies
+        additional_definitions = {
```
Comment (uilianries) [inline conanfile.py]: Use the cmake.definitions as before. It's a pattern, so please, follow it.

File: conanfile.py [hunk 8]
```diff
@@ -62,25 +80,33 @@ def build(self):
         cmake.build()
 
     def package(self):
-        self.copy(pattern="COPYING*", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="COPYING*", dst="licenses",
```
Comment (uilianries) [inline conanfile.py]: be careful with the auto formatting tool. there is no change on this line.

File: conanfile.py [hunk 9]
```diff
@@ -62,25 +80,33 @@ def build(self):
...
-            else ['mongoc-static-1.0', 'bson-static-1.0']
+        if self.options.shared:
+            self.cpp_info.libs = ['mongoc-1.0', 'bson-1.0']
+        else:
+            self.cpp_info.libs = ['mongoc-static-1.0', 'bson-static-1.0']
+
         self.cpp_info.includedirs = [os.path.join("include", "libmongoc-1.0"),
                                      os.path.join("include", "libbson-1.0")]
 
+        if self.options.icu and not self.options.shared:
```
Comment (uilianries) [inline conanfile.py:98]: You could put a comment here (in the code) about this change. It's well explained in your PR.

**Review by uilianries — CHANGES_REQUESTED** [review]

File: conanfile.py [hunk 10]
```diff
@@ -70,23 +86,30 @@ def package(self):
...
                                      os.path.join("include", "libbson-1.0")]
 
+        # If ICU dependency is explicitly set, propagate it
+        if self.options.icu and not self.options.shared:
+            self.cpp_info.libs.extend(self.deps_cpp_info["icu"].libs)
+
         if tools.os_info.is_macos:
-            self.cpp_info.exelinkflags = ['-framework CoreFoundation', '-framework Security']
+            self.cpp_info.frameworks.extend(['CoreFoundation', 'Security'])
             self.cpp_info.sharedlinkflags = self.cpp_info.exelinkflags
```
Comment (Croydon) [inline conanfile.py]:
```suggestion
```

**Review by Croydon — CHANGES_REQUESTED** [review]

File: conanfile.py (hunk 10 above)
Comment (Croydon) [reply conanfile.py]: Leftover. Not needed anymore due to the `cpp_info.frameworks` integration

**Review by uilianries — APPROVED** [review]

**Pull request merged**